digdag-standards/src/main/java/io/digdag/standards/operator/td/BaseTdJobOperator.java

```diff
@@ -0,0 +1,64 @@
+package io.digdag.standards.operator.td;
+
+import com.google.common.base.Optional;
+import io.digdag.client.config.Config;
+import io.digdag.spi.TaskRequest;
+import io.digdag.spi.TaskResult;
+import io.digdag.util.BaseOperator;
+
+import java.nio.file.Path;
+
+abstract class BaseTdJobOperator
+        extends BaseOperator
+{
+    private static final String DONE_JOB_ID = "doneJobId";
+
+    protected final Config state;
+    protected final Config params;
+
+    BaseTdJobOperator(Path workspacePath, TaskRequest request)
+    {
+        super(workspacePath, request);
+
+        this.params = request.getConfig().mergeDefault(
+                request.getConfig().getNestedOrGetEmpty("td"));
+
+        this.state = request.getLastStateParams().deepCopy();
+    }
+
+    @Override
+    public final TaskResult runTask()
+    {
+        try (TDOperator op = TDOperator.fromConfig(params)) {
+
+            Optional<String> doneJobId = state.getOptional(DONE_JOB_ID, String.class);
+            TDJobOperator job;
+            if (!doneJobId.isPresent()) {
```

Comment:
Some operators run jobs in `processJobResult` and we need to keep calling `processJobResult` until it finishes "normally" without a polling exception. The above check is to prevent that the "main" job is then re-run.